### PR TITLE
[Minor|  Fix neural nets regressor shape

### LINKED ISSUE #1554: Neural network future regressor invalid shape in model
**Prerequisites**

* [ x ]

**Describe the bug**

I am trying to build a model with hourly time series data, based on [this tutorial](https://neuralprophet.com/tutorials/tutorial06.html). To my dataframe with 'ds' and 'y', I add a column with the data of the future regressor (temperature).
When fitting the model with neural networks I'm getting a shape error 
RuntimeError: shape '[1024, 24, 1]' is invalid for input of size 7077888

Note: The error doesn't occur when the model is linear 'future_regressors_model = 'linear' ' with future regressor, nor when in the case of the model with neural network but without the future regressor.

**To Reproduce**

1. Start with a df  with the columns 'datetime' ('ds'), 'price'('y'), 'temp' that contains the price of energy and the temperature of a specific city with timestep of one hour for the past 3 years. 
2. Setting the model hyperparameters as follows and splitting the df into train and validation set.
```
m = NeuralProphet(
    yearly_seasonality=False,
    weekly_seasonality=False,
    daily_seasonality=True,
    future_regressors_model = 'neural_nets', # 'linear' default or 'neural_nets'
    future_regressors_d_hidden = 8,# (int)
    future_regressors_num_hidden_layers = 5, # (int)
    n_forecasts=24,
    n_lags=24*11,
    drop_missing=True,
    trainer_config={'accelerator':"gpu"}
)

# Use static plotly in notebooks
m.set_plotting_backend("plotly")

# Add the new future regressor
m.add_future_regressor("temp")

# Add counrty holidays
m.add_country_holidays("IT", mode="additive", lower_window=-1, upper_window=1)
```

4. After running these code lines 'metrics = m.fit(df_train, validation_df=df_val, freq="h", epochs=100, early_stopping=True, progress=False)' ,  this is the error message I get:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[18], line 1
----> 1 metrics = m.fit(df_train, validation_df=df_val, freq="h", epochs=100, early_stopping=True, progress=False)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\forecaster.py:1085, in NeuralProphet.fit(self, df, freq, validation_df, epochs, batch_size, learning_rate, early_stopping, minimal, metrics, progress, checkpointing, continue_training, num_workers)
   1072     df_val = _check_dataframe(self, df_val, check_y=False, exogenous=False)
   1073     df_val = _handle_missing_data(
   1074         df=df_val,
   1075         freq=self.data_freq,
   (...)
   1083         predicting=False,
   1084     )
-> 1085     metrics_df = self._train(
   1086         df,
   1087         df_val=df_val,
   1088         progress_bar_enabled=bool(progress),
   1089         metrics_enabled=bool(self.metrics),
   1090         checkpointing_enabled=checkpointing,
   1091         continue_training=continue_training,
   1092         num_workers=num_workers,
   1093     )
   1095 # Show training plot
   1096 if progress == "plot":

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\forecaster.py:2780, in NeuralProphet._train(self, df, df_val, progress_bar_enabled, metrics_enabled, checkpointing_enabled, continue_training, num_workers)
   2778 self.config_train.set_lr_finder_args(dataset_size=dataset_size, num_batches=len(train_loader))
   2779 # Find suitable learning rate
-> 2780 lr_finder = self.trainer.tuner.lr_find(
   2781     self.model,
   2782     train_dataloaders=train_loader,
   2783     val_dataloaders=val_loader,
   2784     **self.config_train.lr_finder_args,
   2785 )
   2786 # Estimate the optimat learning rate from the loss curve
   2787 assert lr_finder is not None

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\tuner\tuning.py:267, in Tuner.lr_find(self, model, train_dataloaders, val_dataloaders, dataloaders, datamodule, method, min_lr, max_lr, num_training, mode, early_stop_threshold, update_attr)
    264 lr_finder_callback._early_exit = True
    265 self.trainer.callbacks = [lr_finder_callback] + self.trainer.callbacks
--> 267 self.trainer.fit(model, train_dataloaders, val_dataloaders, datamodule)
    269 self.trainer.callbacks = [cb for cb in self.trainer.callbacks if cb is not lr_finder_callback]
    271 self.trainer.auto_lr_find = False

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\trainer.py:608, in Trainer.fit(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    606 model = self._maybe_unwrap_optimized(model)
    607 self.strategy._lightning_module = model
--> 608 call._call_and_handle_interrupt(
    609     self, self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
    610 )

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\call.py:38, in _call_and_handle_interrupt(trainer, trainer_fn, *args, **kwargs)
     36         return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
     37     else:
---> 38         return trainer_fn(*args, **kwargs)
     40 except _TunerExitException:
     41     trainer._call_teardown_hook()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\trainer.py:650, in Trainer._fit_impl(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    643 ckpt_path = ckpt_path or self.resume_from_checkpoint
    644 self._ckpt_path = self._checkpoint_connector._set_ckpt_path(
    645     self.state.fn,
    646     ckpt_path,  # type: ignore[arg-type]
    647     model_provided=True,
    648     model_connected=self.lightning_module is not None,
    649 )
--> 650 self._run(model, ckpt_path=self.ckpt_path)
    652 assert self.state.stopped
    653 self.training = False

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\trainer.py:1097, in Trainer._run(self, model, ckpt_path)
   1095 # hook
   1096 if self.state.fn == TrainerFn.FITTING:
-> 1097     self._call_callback_hooks("on_fit_start")
   1098     self._call_lightning_module_hook("on_fit_start")
   1100 self._log_hyperparams()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\trainer.py:1394, in Trainer._call_callback_hooks(self, hook_name, *args, **kwargs)
   1392     if callable(fn):
   1393         with self.profiler.profile(f"[Callback]{callback.state_key}.{hook_name}"):
-> 1394             fn(self, self.lightning_module, *args, **kwargs)
   1396 if pl_module:
   1397     # restore current_fx when nested context
   1398     pl_module._current_fx_name = prev_fx_name

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\callbacks\lr_finder.py:122, in LearningRateFinder.on_fit_start(self, trainer, pl_module)
    121 def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
--> 122     self.lr_find(trainer, pl_module)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\callbacks\lr_finder.py:107, in LearningRateFinder.lr_find(self, trainer, pl_module)
    105 def lr_find(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
    106     with isolate_rng():
--> 107         self.optimal_lr = lr_find(
    108             trainer,
    109             pl_module,
    110             min_lr=self._min_lr,
    111             max_lr=self._max_lr,
    112             num_training=self._num_training_steps,
    113             mode=self._mode,
    114             early_stop_threshold=self._early_stop_threshold,
    115             update_attr=self._update_attr,
    116         )
    118     if self._early_exit:
    119         raise _TunerExitException()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\tuner\lr_finder.py:247, in lr_find(trainer, model, min_lr, max_lr, num_training, mode, early_stop_threshold, update_attr)
    244 lr_finder._exchange_scheduler(trainer)
    246 # Fit, lr & loss logged in callback
--> 247 _try_loop_run(trainer, params)
    249 # Prompt if we stopped early
    250 if trainer.global_step != num_training + start_steps:

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\tuner\lr_finder.py:481, in _try_loop_run(trainer, params)
    479 loop.load_state_dict(deepcopy(params["loop_state_dict"]))
    480 loop.restarting = False
--> 481 loop.run()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\loop.py:199, in Loop.run(self, *args, **kwargs)
    197 try:
    198     self.on_advance_start(*args, **kwargs)
--> 199     self.advance(*args, **kwargs)
    200     self.on_advance_end()
    201     self._restarting = False

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\fit_loop.py:267, in FitLoop.advance(self)
    265 self._data_fetcher.setup(dataloader, batch_to_device=batch_to_device)
    266 with self.trainer.profiler.profile("run_training_epoch"):
--> 267     self._outputs = self.epoch_loop.run(self._data_fetcher)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\loop.py:200, in Loop.run(self, *args, **kwargs)
    198     self.on_advance_start(*args, **kwargs)
    199     self.advance(*args, **kwargs)
--> 200     self.on_advance_end()
    201     self._restarting = False
    202 except StopIteration:

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\epoch\training_epoch_loop.py:250, in TrainingEpochLoop.on_advance_end(self)
    248 if should_check_val:
    249     self.trainer.validating = True
--> 250     self._run_validation()
    251     self.trainer.training = True
    253 # update plateau LR scheduler after metrics are logged

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\epoch\training_epoch_loop.py:308, in TrainingEpochLoop._run_validation(self)
    305 self.val_loop._reload_evaluation_dataloaders()
    307 with torch.no_grad():
--> 308     self.val_loop.run()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\loop.py:199, in Loop.run(self, *args, **kwargs)
    197 try:
    198     self.on_advance_start(*args, **kwargs)
--> 199     self.advance(*args, **kwargs)
    200     self.on_advance_end()
    201     self._restarting = False

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\dataloader\evaluation_loop.py:152, in EvaluationLoop.advance(self, *args, **kwargs)
    150 if self.num_dataloaders > 1:
    151     kwargs["dataloader_idx"] = dataloader_idx
--> 152 dl_outputs = self.epoch_loop.run(self._data_fetcher, dl_max_batches, kwargs)
    154 # store batch level output per dataloader
    155 self._outputs.append(dl_outputs)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\loop.py:199, in Loop.run(self, *args, **kwargs)
    197 try:
    198     self.on_advance_start(*args, **kwargs)
--> 199     self.advance(*args, **kwargs)
    200     self.on_advance_end()
    201     self._restarting = False

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\epoch\evaluation_epoch_loop.py:137, in EvaluationEpochLoop.advance(self, data_fetcher, dl_max_batches, kwargs)
    134 self.batch_progress.increment_started()
    136 # lightning module methods
--> 137 output = self._evaluation_step(**kwargs)
    138 output = self._evaluation_step_end(output)
    140 self.batch_progress.increment_processed()

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\loops\epoch\evaluation_epoch_loop.py:234, in EvaluationEpochLoop._evaluation_step(self, **kwargs)
    223 """The evaluation step (validation_step or test_step depending on the trainer's state).
    224 
    225 Args:
   (...)
    231     the outputs of the step
    232 """
    233 hook_name = "test_step" if self.trainer.testing else "validation_step"
--> 234 output = self.trainer._call_strategy_hook(hook_name, *kwargs.values())
    236 return output

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\trainer\trainer.py:1494, in Trainer._call_strategy_hook(self, hook_name, *args, **kwargs)
   1491     return
   1493 with self.profiler.profile(f"[Strategy]{self.strategy.__class__.__name__}.{hook_name}"):
-> 1494     output = fn(*args, **kwargs)
   1496 # restore current_fx when nested context
   1497 pl_module._current_fx_name = prev_fx_name

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\pytorch_lightning\strategies\strategy.py:390, in Strategy.validation_step(self, *args, **kwargs)
    388 with self.precision_plugin.val_step_context():
    389     assert isinstance(self.model, ValidationStep)
--> 390     return self.model.validation_step(*args, **kwargs)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\time_net.py:817, in TimeNet.validation_step(self, batch, batch_idx)
    815     meta_name_tensor = None
    816 # Run forward calculation
--> 817 predicted, _ = self.forward(inputs, meta_name_tensor)
    818 # Calculate loss
    819 loss, reg_loss = self.loss_func(inputs, predicted, targets)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\time_net.py:603, in TimeNet.forward(self, inputs, meta, compute_components_flag)
    601 if "regressors" in inputs:
    602     if "additive" in inputs["regressors"].keys():
--> 603         additive_regressors = self.future_regressors(inputs["regressors"]["additive"], "additive")
    604         additive_components_nonstationary += additive_regressors
    605         components["additive_regressors"] = additive_regressors

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\torch\nn\modules\module.py:1511, in Module._wrapped_call_impl(self, *args, **kwargs)
   1509     return self._compiled_call_impl(*args, **kwargs)  # type: ignore[misc]
   1510 else:
-> 1511     return self._call_impl(*args, **kwargs)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\torch\nn\modules\module.py:1520, in Module._call_impl(self, *args, **kwargs)
   1515 # If we don't have any hooks, we want to skip the rest of the logic in
   1516 # this function, and just call forward.
   1517 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1518         or _global_backward_pre_hooks or _global_backward_hooks
   1519         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1520     return forward_call(*args, **kwargs)
   1522 try:
   1523     result = None

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\components\future_regressors\neural_nets.py:128, in NeuralNetsFutureRegressors.forward(self, inputs, mode, indeces)
    114 """Compute all seasonality components.
    115 Parameters
    116 ----------
   (...)
    124         Forecast component of dims (batch, n_forecasts, no_quantiles)
    125 """
    127 if "additive" == mode:
--> 128     f_r = self.all_regressors(inputs, mode="additive")
    129 if "multiplicative" == mode:
    130     f_r = self.all_regressors(inputs, mode="multiplicative")

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\components\future_regressors\neural_nets.py:108, in NeuralNetsFutureRegressors.all_regressors(self, regressor_inputs, mode)
    106 regressor_input = regressor_inputs[:, :, regressor_index].unsqueeze(dim=2)
    107 if i == 0:
--> 108     x = self.regressor(regressor_input, name=name)
    109 if i > 0:
    110     x = x + self.regressor(regressor_input, name=name)

File c:\Users\username\AppData\Local\anaconda3\envs\NeuralProphet\Lib\site-packages\neuralprophet\components\future_regressors\neural_nets.py:88, in NeuralNetsFutureRegressors.regressor(self, regressor_input, name)
     85     x = self.regressor_nets[name][i](x)
     87 # segment the last dimension to match the quantiles
---> 88 x = x.reshape(x.shape[0], self.n_forecasts, len(self.quantiles))
     89 return x

RuntimeError: shape '[1024, 24, 1]' is invalid for input of size 7077888
```

**Expected behavior**

I want to fit the model.

**What actually happens**

See above; this happens every time I try to run the code.

**Environment (please complete the following information):**

 - Python 3.11.8
 - NeuralProphet 0.8.0 (installed with `pip install neuralprophet`)

### PR DESCRIPTION
**Pull request by MaiBe-ctrl**
## :microscope: Background

- Fixes #1554

## :crystal_ball: Key changes

- Explain the main changes introduced by this pull request for the reviewer.

## :clipboard: Review Checklist
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, added docstrings and data types to function definitions.
- [ ] I have added pytests to check whether my feature / fix works.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: neuralprophet/components/future_regressors/neural_nets.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def __init__(self, config, id_list, quantiles, n_forecasts, device, config_trend
                     regressor_net.append(nn.Linear(d_inputs, self.d_hidden_regressors, bias=True))
                     d_inputs = self.d_hidden_regressors
                 # final layer has input size d_inputs and output size equal to no. of forecasts * no. of quantiles
```
Comment (ourownstory) [inline neuralprophet/components/future_regressors/neural_nets.py:36]: ^ this comment may need updating. 

Q: Is this NN truly applied separately to each observation and each forecast target?

Comment (MaiBe-ctrl) [reply neuralprophet/components/future_regressors/neural_nets.py:36]: It's been applied for each batch layer by layer

**Review by ourownstory — CHANGES_REQUESTED** [review]
If the removal of multiplying the outputs with `n_forecasts` addresses the problem, there may be more places where this needs to be addressed. 
Essentially all of the files in `components/future_regressors` need to be screened for this, especially any with NN, but also auxiliary functions. 
It seems like these code areas lack tests - we should add those, too.

**Review by ourownstory — APPROVED** [review]
Fixed shared NN and partially fixed shared nn coef, created new bug report.

**Pull request merged**